src/System.Management.Automation/engine/parser/TypeInferenceVisitor.cs

```diff
@@ -1021,8 +1021,24 @@ private void InferTypesFrom(CommandAst commandAst, List<PSTypeName> inferredType
             PseudoBindingInfo pseudoBinding = new PseudoParameterBinder()
             .DoPseudoParameterBinding(commandAst, null, null, PseudoParameterBinder.BindingType.ParameterCompletion);
 
-            if (pseudoBinding?.CommandInfo == null)
+            if (pseudoBinding?.CommandInfo is null)
             {
+                var commandName = commandAst.GetCommandName();
+                try
+                {
+                    var foundCommand = _context.ExecutionContext.CommandDiscovery.LookupCommandInfo(commandName);
```

Comment:
How about change to the following to specify the command type, so that the lookup can be a little faster?

```suggestion
                    var foundCommand = CommandDiscovery.LookupCommandInfo(
                        commandName,
                        CommandTypes.Application,
                        SearchResolutionOptions.ResolveLiteralThenPathPatterns,
                        CommandOrigin.Internal,
                        _context);
```